src/nvim/event/pty_process.c

```diff
@@ -132,6 +132,9 @@ static void init_child(PtyProcess *ptyproc) FUNC_ATTR_NONNULL_ALL
   signal(SIGTERM, SIG_DFL);
   signal(SIGALRM, SIG_DFL);
 
+  if (proc->cwd)
+    os_chdir((char *)proc->cwd);
```

Comment:
Isn't this going to change CWD for the entire nvim process? Where is this reversed?